eth/catalyst/api.go

```diff
@@ -769,7 +865,7 @@ func (api *ConsensusAPI) ExecuteStatelessPayloadV4(params engine.ExecutableData,
 	return api.executeStatelessPayload(params, versionedHashes, beaconRoot, requests, opaqueWitness)
 }
 
-func (api *ConsensusAPI) newPayload(params engine.ExecutableData, versionedHashes []common.Hash, beaconRoot *common.Hash, requests [][]byte, witness bool) (engine.PayloadStatusV1, error) {
+func (api *ConsensusAPI) newPayload(params engine.ExecutableData, versionedHashes []common.Hash, beaconRoot *common.Hash, requests [][]byte, inclusionList engine.InclusionList, witness bool) (engine.PayloadStatusV1, error) {
```

Comment:
The parameters here become quite clunky. Maybe we should create a struct here to pass the data, makes it easier to add fields in the future

Reply:
Fair point. But until we get into this implementation seriously, I'd like to keep it simple and naive as `master` can be refactored. I believe applying these changes into `master` when the time comes would be cleaner.